client/layout/masterbar/checkout.tsx

```diff
@@ -54,6 +55,10 @@ const CheckoutMasterbar = ( {
 	const { responseCart, replaceProductsInCart } = useShoppingCart( cartKey );
 	const [ isModalVisible, setIsModalVisible ] = useState( false );
 
+	const isPremiumSupportEligible = responseCart.products.some( ( product ) =>
+		PREMIUM_SUPPORT_PRODUCT_SLUGS.includes( product.product_slug )
+	);
```

Comment:
I would move this logic inside `DefaultMasterbarContact` to avoid prop drilling.